Training/Training/TestCase.c

```diff
@@ -61,18 +69,40 @@ void main () {
       printf ("Test Case%d: Input--> %d ", i + 1, num2);
       char* binResult2 = DecToHex (num2);
       int n = strcmp (binResult2, output2[i]);
-      if (n == 0) printf ("\033[1;32mPASS\033[0m\n");
-      else printf ("\033[1;31mFAIL\033[0m\n");
+      if (n == 0) printf (ANSI_COLOR_GREEN"PASS\n"ANSI_RESET_ALL);
+      else printf (ANSI_COLOR_RED"FAIL\n"ANSI_RESET_ALL);
    }
    //checking for decimal numbers exceeding integer range
    long long int inputLong2 = -2147483649ll;
    char* outputLong2 = "FFFFFFFF7FFFFFFF";
    printf ("Test Case10: Input--> %lld ", inputLong2);
    char* binResultLong2 = DecToHex (inputLong2);
    int l = strcmp (binResultLong2, outputLong2);
-   if (l == 0) printf ("\033[1;32mPASS\033[0m\n");
+   if (l == 0) printf (ANSI_COLOR_GREEN"PASS\n"ANSI_RESET_ALL);
    else {
-      printf ("\033[1;31mFAIL\033[0m\n");
-      printf ("Expected: \033[1;35m%s\033[0m Output: \033[1;31m%s\033[0m\n", outputLong2, binResultLong2);
+      printf (ANSI_COLOR_RED"FAIL\n"ANSI_RESET_ALL);
+      printf ("Expected: "ANSI_COLOR_MAGENTA"%s"ANSI_RESET_ALL" Output: "ANSI_COLOR_RED"%s\n"ANSI_RESET_ALL, outputLong2, binResultLong2);
+   }
+   UserInput (); //Validating user input values
+}
+
+int UserInput () {
+   int dec;
+   char term;
+   while (1) {
+      printf ("\nEnter a decimal number:");
+      if (scanf_s ("%d%c", &dec, &term, 1) != 2 || term != '\n') {
+         printf (ANSI_COLOR_RED"Please enter a VALID NUMBER!!!\n"ANSI_RESET_ALL);
+         for (;;) {
+            term = fgetc (stdin);
+            if (term == EOF || term == '\n') break;
+         }
+      } else {
+         char* binNum = DecToBin (dec);
+         printf ("Binary value(0b): "ANSI_COLOR_BLUE"%s"ANSI_RESET_ALL, binNum);
+         char* hexNum = DecToHex (dec);
+         printf ("\nHexadecimal value(0x): "ANSI_COLOR_MAGENTA"%s\n"ANSI_RESET_ALL, hexNum);
+      }
    }
+   return 0;

```

Comment:
- Add two method named TestDec2Bin and TestDec2Hex
- Add these cases 
![image](https://github.com/user-attachments/assets/d16258ce-7bc1-43f9-9587-d810b2efcbcb)
Make it as two separate methods